src/easylink/pipeline_graph.py

```diff
@@ -77,26 +77,42 @@ def any_embarrassingly_parallel(self) -> bool:
         to be run in an embarrassingly parallel way."""
         return any(
             [
-                self.get_whether_embarrassingly_parallel(node)
+                self.get_embarrassingly_parallel_details(node)[0]
                 for node in self.implementation_nodes
             ]
         )
 
-    def get_whether_embarrassingly_parallel(self, node: str) -> bool:
+    def get_embarrassingly_parallel_details(self, node: str) -> tuple[bool, bool, bool]:
```

Comment:
I dislike the boolean tuple here, would prefer a dictionary or dataclass for readability